spring-context/src/main/java/org/springframework/cache/interceptor/CacheAspectSupport.java

```diff
@@ -1201,8 +1202,8 @@ private class ReactiveCachingHandler {
 				}
 				if (adapter.isMultiValue()) {
 					return adapter.fromPublisher(Flux.from(Mono.fromFuture(cachedFuture))
-							.switchIfEmpty(Flux.defer(() -> (Flux) evaluate(null, invoker, method, contexts)))
-							.flatMap(v -> evaluate(valueToFlux(v, contexts), invoker, method, contexts))
+							.switchIfEmpty(Flux.defer(() -> (Flux) Objects.requireNonNull(evaluate(null, invoker, method, contexts))))
```

Comment:
For code on the hot path, in reactor-core, I used assertions to silence the warnings where I knew nullable API is not returning a null value. I wasn't sure what strategy you'd like to take so I left a TODO somewhere near `requireNonNull` to address this.